syntax = "proto3";
package instance_relation;





/*
InstanceRelationSnapshot请求
*/
message InstanceRelationSnapshotRequest {
    /*
    关系ID
    */
    string relation_id = 1 ;
    /*
    关系实例ID 关系id+左右实例ID 
    */
    string relation_instance_id = 2 ;
    /*
    操作时间
    */
    int32 _ts = 3 ;
}

/*
InstanceRelationSnapshot返回
*/
message InstanceRelationSnapshotResponse {
message XLeftInstance {
    /*
    实例ID
    */
    string instanceId = 1 ;
    /*
    主机名
    */
    string hostname = 2 ;
    /*
    实例名称
    */
    string name = 3 ;
    /*
    模型ID
    */
    string _object_id = 4 ;
    /*
    实例版本号
    */
    int32 _version = 5 ;
}
        
message XRightInstance {
    /*
    实例ID
    */
    string instanceId = 1 ;
    /*
    主机名
    */
    string hostname = 2 ;
    /*
    实例名称
    */
    string name = 3 ;
    /*
    模型ID
    */
    string _object_id = 4 ;
    /*
    实例版本号
    */
    int32 _version = 5 ;
}
        
    
    /*
    快照ID
    */
    string _id = 1 ;
    /*
    左端实例数据
    */
    XLeftInstance _left_instance = 2 ;
    /*
    操作类型
    */
    string _operation = 3 ;
    /*
    操作用户
    */
    string _operator = 4 ;
    /*
    关系ID
    */
    string _relation_id = 5 ;
    /*
    右端实例数据
    */
    XRightInstance _right_instance = 6 ;
    /*
    操作时间
    */
    int32 _ts = 7 ;
    /*
    版本号
    */
    int32 _version = 8 ;
    /*
    左端实例ID
    */
    string left_instance_id = 9 ;
    /*
    右端实例ID
    */
    string right_instance_id = 10 ;
    /*
    关系实例ID
    */
    string relation_instance_id = 11 ;
}

/*
InstanceRelationSnapshotApi返回
*/
message InstanceRelationSnapshotResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    instance_relation.InstanceRelationSnapshotResponse data = 4 ;
}

